src/main/java/io/servertap/api/v1/ServerApi.java

```diff
@@ -206,6 +204,111 @@ public static void saveWorld(Context ctx) {
         ctx.json("success");
     }
 
+    private static void addFolderToZip(File folder, ZipOutputStream zip, String baseName, String rootName) throws IOException {
+        File[] files = folder.listFiles();
+        assert files != null;
+        for (File file : files) {
+            if (file.isDirectory()) {
+                if(rootName == null) {
+                    addFolderToZip(file, zip, baseName, folder.getName());
+                } else {
+                    addFolderToZip(file, zip, baseName, rootName);
+                }
+            } else {
+                String name = file.getAbsolutePath().substring(baseName.length())
+                        // Trim first slash (absolute path)
+                        .replaceFirst("^/", "");
+//              Join path and folder name
+                if(rootName == null) {
+                    name = folder.getName() + "/" + name;
+                } else {
+                    name = rootName + "/" + name;
+                }
+                ZipEntry zipEntry = new ZipEntry(name);
+                zip.putNextEntry(zipEntry);
+                IOUtils.copy(new FileInputStream(file), zip);
+                zip.closeEntry();
+            }
+        }
+    }
+
+    @OpenApi(
+            path = "/v1/worlds/:uuid/download",
+            summary = "Downloads a Zip compressed archive of the world's folder",
+            method = HttpMethod.GET,
+            tags = {"Server"},
+            headers = {
+                    @OpenApiParam(name = "key")
+            },
+            pathParams = {
+                    @OpenApiParam(name = "uuid", description = "The UUID of the World to download")
+            },
+            responses = {
+                    @OpenApiResponse(status = "200", content = @OpenApiContent(type = "application/zip")),
+            }
+    )
+    public static void downloadWorld(Context ctx) throws IOException {
+        org.bukkit.Server bukkitServer = Bukkit.getServer();
+
+        UUID worldUUID = ValidationUtils.safeUUID(ctx.pathParam("uuid"));
+        if (worldUUID == null) {
+            throw new BadRequestResponse(Constants.INVALID_UUID);
+        }
+
+        org.bukkit.World world = bukkitServer.getWorld(worldUUID);
+
+        if (world != null) {
+            File folder = world.getWorldFolder();
+
+            // Set headers for proper zip download
+            ctx.header("Content-Disposition", "attachment; filename=\"" + folder.getName() + ".zip\"");
+            ctx.header("Content-Type", "application/zip");
+
+
+            ZipOutputStream zip = new ZipOutputStream(ctx.res.getOutputStream());
+            addFolderToZip(folder, zip, folder.getAbsolutePath(), null);
+            zip.close();
+        }
```

Comment:
Here there should be an `else` for when `world` is `null`, with just `throw new NotFoundResponse(Constants.WORLD_NOT_FOUND);` in it.